libfreerdp/core/gateway/http.c

```diff
@@ -1016,12 +1362,18 @@ HttpResponse* http_response_new(void)
 	if (!response->Authenticates)
 		goto fail;
 
+	response->SetCookie = ListDictionary_New(FALSE);
+
+	if (!response->SetCookie)
```

Comment:
better add a new/free function to this. otherwise you need to guarantee that the cookie string is not stack allocated and the lifecycle the same as this.

Reply:
as far as I understood every string in HttpResponse is only valid as long as HttpResponse is around (every string essentially points to the response buffer read over tls/response->data). This is why i allocate/deallocate in the http_context_set_cookie case (because there I need to guarantee this string is valid for the context's lifetime)